src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp

```diff
@@ -16,170 +15,157 @@
-            //        RaiseDead_Timer = 30000;
-            SummonSkeletons_Timer = 34000;
-        }
+        events.ScheduleEvent(EVENT_SHADOWBOLT, 5s);
+        events.ScheduleEvent(EVENT_CLEAVE, 8s);
+        events.ScheduleEvent(EVENT_MORTALSTRIKE, 12s);
+        events.ScheduleEvent(EVENT_RAISE_DEAD, 15s);
 
-        InstanceScript* instance;
+        BossAI::JustEngagedWith(who);
+    }
 
-        uint32 ShadowBolt_Timer;
-        uint32 Cleave_Timer;
-        uint32 MortalStrike_Timer;
-        //    uint32 RaiseDead_Timer;
-        uint32 SummonSkeletons_Timer;
+    void JustDied(Unit* killer) override
+    {
+        // needed until re-write of instance scripts is done
+        instance->SetData(TYPE_BARON, DONE);
 
-        void Reset() override
-        {
-            Initialize();
-            if (instance->GetData(TYPE_RAMSTEIN) == DONE)
-                instance->SetData(TYPE_BARON, NOT_STARTED);
-        }
+        BossAI::JustDied(killer);
+    }
 
-        void AttackStart(Unit* who) override
-        {
-            //can't use entercombat(), boss' dmg aura sets near players in combat, before entering the room's door
-            if (instance->GetData(TYPE_BARON) == NOT_STARTED)
-                instance->SetData(TYPE_BARON, IN_PROGRESS);
-            ScriptedAI::AttackStart(who);
-        }
+    void UpdateAI(uint32 diff) override
+    {
+        if (!UpdateVictim())
+            return;
 
-        void JustSummoned(Creature* summoned) override
-        {
-            if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0))
-                summoned->AI()->AttackStart(target);
-        }
+        events.Update(diff);
 
-        void JustDied(Unit* /*killer*/) override
-        {
-            instance->SetData(TYPE_BARON, DONE);
-        }
+        if (me->HasUnitState(UNIT_STATE_CASTING))
+            return;
 
-        void UpdateAI(uint32 diff) override
+        while (uint32 eventId = events.ExecuteEvent())
         {
-            if (!UpdateVictim())
-                return;
-
-            //ShadowBolt
-            if (ShadowBolt_Timer <= diff)
+            switch (eventId)
             {
-                if (SelectTarget(SELECT_TARGET_RANDOM, 0))
-                    DoCastVictim(SPELL_SHADOWBOLT);
+                case EVENT_SHADOWBOLT:
+                    if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0))
+                        DoCast(target, SPELL_SHADOWBOLT);
+
+                    events.Repeat(10s);
+                    break;
+                case EVENT_CLEAVE:
+                    DoCastVictim(SPELL_CLEAVE);
+                    events.Repeat(7s, 17s);
+                    break;
+                case EVENT_MORTALSTRIKE:
+                    DoCastVictim(SPELL_MORTALSTRIKE);
+                    events.Repeat(10s, 25s);
+                    break;
+                case EVENT_RAISE_DEAD:
+                    if (!RaiseDead)
+                    {
+                        DoCast(SPELL_RAISE_DEAD);
```

Comment:
DoCastSelf please.
DoCast will force core chose a target, and this case target will be caster, so is better explicit it :)